### ISSUE TITLE
`wrapCompilerAsTypeGuard` and `wrapValidatorAsTypeGuard` are not usable in Deno build

### ISSUE THREAD

**Issue by curtislarson** [ThomasAribart/json-schema-to-ts#74]
Hello! First off thanks for writing this library! It's awesome and just keeps getting better!

One thing I noticed is in the Deno build the wrapper functions cannot be used because its packaged as only type definitions and the implementations are stripped by rollup

https://github.com/ThomasAribart/json-schema-to-ts/blob/d8faafb4603aab1e0496858bd727f591940cc137/builds/deno/index.d.ts#L307

https://github.com/ThomasAribart/json-schema-to-ts/blob/d8faafb4603aab1e0496858bd727f591940cc137/builds/deno/index.d.ts#L312

Trying to import it and use it throws

```
error: SyntaxError: The requested module 'https://deno.land/x/json_schema_to_ts@v2.5.1/index.d.ts' does not provide an export named 'wrapCompilerAsTypeGuard'
```

I fixed it by just adding the implementation real quick and changing it to a `.ts` file since Deno doesnt always play nice with `.d.ts` files. That may not work the best with your rollup setup though and im sure you know a more compatible fix!

**Comment by ThomasAribart** [comment]
Hi @curtislarson ! Good catch, thanks for the heads up.

It seems that I now export more than types but forgot to modify the rollup configuration for deno export: https://github.com/ThomasAribart/json-schema-to-ts/blob/master/rollup.config.js

I'm gonna look into that !

**Comment by ThomasAribart** [comment]
Hi @curtislarson ! I found https://www.npmjs.com/package/@rollup/plugin-typescript which seems to bundle the typescript in cjs or esm.

I could use it along with dts to export an `index.js` in parallel to the `index.d.ts`. BUT I'm not sure if this is the best approach. Wouldn't it be better to simply export a single `index.ts` file ? Sorry, I'm not familiar with Deno.

**Comment by curtislarson** [comment]
@ThomasAribart The `.js` and `.d.ts` is a valid approach that some libraries like [esbuild](https://github.com/esbuild/deno-esbuild/blob/c04762b596b09d750f668ae3f6e10799b3470c61/mod.js#L1) take with the `<reference types` notation.

I think most Deno people prefer importing from `ts` files since they may not be aware of the typing inclusion in the `js` file or would have to do the awkward `@deno-types` annotation. Definitely make sure it's ESM though and people can import it directly from github / unpkg / denoland / esm etc

Although honestly it feels like there will be a shift back to `js` extensions soon with typescript never allowing `.ts` imports and the shift away from typechecking in Deno. That's a bit too off topic though :)

**Comment by curtislarson** [comment]
https://gist.github.com/curtislarson/0253736b428c641411183cc41dec3202 This is pretty much what the file I use looks like, where you just `import type`/`export type` instead of `import`/`export` and include the function definitions.

Although you can go even further and use the optimized pinned urls from skypack...

**Comment by ThomasAribart** [comment]
@curtislarson Yes, that's what I had in mind. It sounds easy but it is in fact quite hard to automate :/ I couldn't find a rollup plugin that flattens ts like that. It seems simpler to export it as `d.ts` on one side and `.js` or `.mjs` on another.

**Comment by curtislarson** [comment]
I haven't experimented with them a lot but there are packages like [unplugin](https://github.com/unjs/unplugin) and [unbuild](https://github.com/unjs/unbuild) that seem to give you a lot of flexibility and additional options when it comes to generating output with rollup, esbuild, etc.

**Comment by ThomasAribart** [comment]
Sorry, those plugin don't allow to flatten ts filte easily either.

I've release a beta of v2.5.2 with the same `d.ts` as usual, but with `js` and `mjs` rollups containing both wrappers in `cjs` and `esm`. Can you try those and tell me if it works for you ? If it does, I'll officially release the 2.5.2.

https://deno.land/x/json_schema_to_ts@v2.5.2-beta.0

**Comment by curtislarson** [comment]
@ThomasAribart I'm able to import them but have to manually specify the correct typings via `@deno-types` in order to get it working.

<img width="601" alt="Screen Shot 2022-05-26 at 12 44 41 PM" src="https://user-images.githubusercontent.com/2141050/170535178-373c76da-e306-44d7-bba3-1a03bfb92b72.png">
<img width="677" alt="Screen Shot 2022-05-26 at 12 44 51 PM" src="https://user-images.githubusercontent.com/2141050/170535179-ea09a369-10d4-4f02-a3df-e9a2da96ab3a.png">

Which you can resolve by just reference the types in the `.mjs`/`.js` file

<img width="1076" alt="Screen Shot 2022-05-26 at 12 45 56 PM" src="https://user-images.githubusercontent.com/2141050/170535278-102709ea-7cba-45b7-b772-94dd964cc817.png">

**Comment by ThomasAribart** [comment]
The [banner option](https://rollupjs.org/guide/en/#outputbanneroutputfooter) seems to have done the job 👍

Can you try the second beta ? https://deno.land/x/json_schema_to_ts@v2.5.2-beta.1

**Comment by curtislarson** [comment]
@ThomasAribart Yep this code worked for me

```typescript
import {
  wrapCompilerAsTypeGuard,
  $Compiler,
  JSONSchema,
} from "https://deno.land/x/json_schema_to_ts@v2.5.2-beta.1/index.mjs";

import Ajv from "https://esm.curtis.land/ajv@8.11.0/dist/2020";

const instance = new Ajv();

const $compile: $Compiler = (schema: JSONSchema) => instance.compile(schema);

const wrapped = wrapCompilerAsTypeGuard($compile);

const validate = wrapped({
  type: "object",
  required: ["name"],
  properties: { name: { type: "string" } },
} as const);

const res = validate({ name: "Curtis" });
console.log(res); // true
```

I don't think the `.js` version will ever be compatible with Deno though unless `exports` is shimmed but the `.mjs` worked great.

**Comment by ThomasAribart** [comment]
Awesome 😎 

Does the type guard work as well ?

```
if (validate(data)) {
  ... // data should be correctly typed here
}
```

If so, I'll release and close this issue tomorrow 👍

Cheers !

**Comment by curtislarson** [comment]
Yep it works perfect. Thanks for the fixes!
